docs/source/executing-operations/file-uploads.mdx

```diff
@@ -0,0 +1,420 @@
+
+Learn how to configure the Apollo Router to receive file uploads in client requests using the [GraphQL multipart request specification](https://github.com/jaydenseric/graphql-multipart-request-spec).
+
+## About file uploads using multipart requests
+
+A [multipart HTTP request](https://www.w3.org/Protocols/rfc1341/7_2_Multipart.html) lets you efficiently send multiple files of various data formats&mdash;such as text, binary, and JSON objects&mdash;in a single HTTP request. The [GraphQL multipart request spec](https://github.com/jaydenseric/graphql-multipart-request-spec) uses multipart requests to upload files using arguments in a GraphQL mutation.
+
+### Example usage
+
+Imagine you're building a platform where users can create posts with a title and an image file.
+Your subgraph schema may include something like this:
+
+```graphql showLineNumbers=false  disableCopy
+type Post {
+  id: ID!
+  title: String!
+  image: Upload!
+}
+
+type Mutation {
+  createPost(title: String!, image: Upload!): Post!
+}
+```
+
+<Note>
+
+Some GraphQL server implementations provide built-in support for handling file uploads, including an `Upload` scalar type.
+For others, including the latest version of [Apollo Server](/apollo-server/), you must use external packages, such as [`graphql-upload`](https://www.npmjs.com/package/graphql-upload).
+Refer to your subgraph library or package documentation for further information, including writing resolvers for uploaded files.
+
+</Note>
+
+When a client calls the `createPost` mutation, it can use variables to include the actual image file to upload:
+
+```graphql showLineNumbers=false  disableCopy
+{
+  query: `
+    mutation CreatePost($title: String!, $image: Upload!) {
+      createPost(title: $title, image: $image) {
+        id
+        title
+        image
+      }
+    }
+  `,
+  variables: {
+    title: "My first post",
+    image: File // image.png
+  }
+}
+```
+
+A request using the GraphQL multipart request spec would include the following as separate parts in a multipart request:
+
+- the above operation definition
+- the image file to upload
+- a map between variables and files to upload
+
+The exact requirements are documented in [Client usage requirements](#client-usage-requirements).
+
+## File upload configuration and usage
+
+To enable file uploads from clients, you must both [configure support in the Apollo Router](#configure-file-upload-support-in-the-router) and [ensure client usage conforms to requirements](#client-usage-requirements).
+
+### Configure file upload support in the router
+
+By default, receiving client file uploads isn't enabled in the Apollo Router.
+To enable file upload support, set the following fields in your `router.yaml` configuration file:
+
+```yaml title="router.yaml" showLineNumbers="false"
+preview_file_uploads:
+  enabled: true
+  protocols:
+    multipart:
+      enabled: true
+      mode: stream
+      limits:
+        max_file_size: 1mb
+        max_files: 10
```

Comment:
This `router.yaml` appears to be causing [this error](https://app.circleci.com/pipelines/github/apollographql/router/21502/workflows/c71c8bba-2fc0-4501-b86f-d88b98cf7976/jobs/147099)

```
--- STDERR:              apollo-router configuration::tests::validate_project_config_files ---
thread 'configuration::tests::validate_project_config_files' panicked at apollo-router/src/configuration/tests.rs:456:21:
../docs/source/executing-operations/file-uploads.mdx configuration error: 
failed to parse yaml: mapping values are not allowed in this context at line 2 column 21
```

However, this config matches what I could find in the `[default.router.yaml](https://github.com/apollographql/router/blob/99ad27a2f749d5d3a311014a2dc215cdfde7b240/apollo-router/tests/fixtures/file_upload/default.router.yaml#L4)` and `[file_uploads.router.yaml](https://github.com/apollographql/router/blob/99ad27a2f749d5d3a311014a2dc215cdfde7b240/apollo-router/src/configuration/testdata/metrics/file_uploads.router.yaml#L4)` exemplars.